fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQuery.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+
+/**
+ *
+ */
+public abstract class SearchQuery {
+    private static final String CLASSNAME = SearchQuery.class.getName();
+    private static final Logger logger = Logger.getLogger(CLASSNAME);
+    private final String rootResourceType;
+    private final List<SearchParam> searchParams = new ArrayList<>();
+
+    // A list of non-parameter related extensions (e.g. location processing)
+    private final List<SearchExtension> extensions = new ArrayList<>();
+
+    /**
+     * Public constructor
+     * @param rootResourceType
+     */
+    public SearchQuery(String rootResourceType) {
+        this.rootResourceType = rootResourceType;
+    }
+
+    /**
+     * Getter for the root resource type for this search query
+     * @return
+     */
+    public String getRootResourceType() {
+        return this.rootResourceType;
+    }
+
+    /**
+     * Add the search parameter to the model
+     * @param sp
+     */
+    public void add(SearchParam sp) {
+        this.searchParams.add(sp);
+    }
+
+    /**
+     * Add the search extension to the model
+     * @param ext
+     */
+    public void add(SearchExtension ext) {
+        this.extensions.add(ext);
+    }
+
+    /**
+     * Get the root query and attach the parameter filters to it
+     * @param <T>
```

Comment:
pattern, do we include <T> in javadoc

Reply:
It gets added automatically by eclipse when adding javadoc for a new method.